fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/service/InstanceTypeServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.configuration.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.fineract.infrastructure.configuration.data.InstanceTypeData;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class InstanceTypeServiceImpl implements InstanceTypeService {
+
+    private final InstanceTypeData instanceTypeData;
+
+    @Override
+    public boolean isReadMode() {
+        return instanceTypeData.isReadModeEnabled();
+    }
+
+    @Override
+    public boolean isWriteMode() {
+        return instanceTypeData.isWriteModeEnabled();
+    }
+
+    @Override
+    public boolean isBatchMode() {
+        return instanceTypeData.isBatchModeEnabled();
+    }
+
+    @Override
+    public boolean isReadOnlyMode() {
+        return instanceTypeData.isReadModeEnabled() && !instanceTypeData.isWriteModeEnabled() && !instanceTypeData.isBatchModeEnabled();
+    }
+
+    @Override
+    public boolean isInstanceTypeConfigured() {
```

Comment:
We could do this check immediately on boot in a Java configuration class (similar to the way that we check that an authentication scheme is configured) and fail early. Injectable service for that not really necessary.